src/Http/Http.Abstractions/src/Internal/ParsingHelpers.cs

```diff
@@ -19,22 +19,22 @@ public static StringValues GetHeader(IHeaderDictionary headers, string key)
         public static StringValues GetHeaderSplit(IHeaderDictionary headers, string key)
         {
             var values = GetHeaderUnmodified(headers, key);
-            return new StringValues(GetHeaderSplitImplementation(values).ToArray());
-        }
 
-        private static IEnumerable<string> GetHeaderSplitImplementation(StringValues values)
-        {
+            StringValues result = default;
+
             foreach (var segment in new HeaderSegmentCollection(values))
             {
                 if (!StringSegment.IsNullOrEmpty(segment.Data))
                 {
                     var value = DeQuote(segment.Data.Value);
                     if (!string.IsNullOrEmpty(value))
                     {
-                        yield return value;
+                        result = StringValues.Concat(in result, value);
```

Comment:
Shouldn't use `in` now that `StringValues` is only a pointer size https://github.com/aspnet/Extensions/pull/1283

OTOH there isn't currently a Concat overload that takes `(StringValues values, string value)` without the `in` so may also want to add that?

Reply:
We can look at this in future changes. How much would it help?